pages/api/roles/index.ts

```diff
@@ -68,12 +69,35 @@ function handler(req: NextApiRequest, res: NextApiResponse) {
   }
 }
 
-// TODO: Optional Search query. Will be done in another PR to unblock others.
-
 async function handleGet(req: NextApiRequest, res: NextApiResponse<ApiResponse<RoleListData[]>>) {
+  const { companyId, searchQuery } = req.query;
+
+  const queryCompanyId = isInteger(companyId as string) ? Number(companyId) : undefined;
+
+  const searchWords = typeof searchQuery === 'string' ? (searchQuery as string).split(' ') : [];
+  const queryYear = searchWords
+    .filter(isInteger)
+    .map(Number)
+    .filter((year) => year >= MIN_ROLE_YEAR)[0];
+  const queryWords = searchWords
+    .filter((word) => !(isInteger(word) && Number(word) >= MIN_ROLE_YEAR))
+    .map((word) => ({
+      title: {
+        contains: word,
+      },
+    }));
+
+  if (!queryCompanyId || !searchWords.length) {
+    res.status(HttpStatus.OK).json(createJsonResponse([]));
+    return;
+  }
+
   const roles: RoleListData[] = await prisma.role.findMany({
     where: {
       isVerified: true,
+      companyId: queryCompanyId,
+      AND: [{ year: { equals: queryYear } }],
+      OR: [...queryWords],
```

Comment:
```suggestion
      OR: queryWords.length > 0 ? queryWords : undefined,
```

> OR
One or more conditions must return true.

In case there are no query words (e.g. `searchString` is `2023`, we should set it to undefined. If not, we will never get any matches.

```sql
prisma:query SELECT "public"."roles"."id", "public"."roles"."title", "public"."roles"."type", 
"public"."roles"."year", "public"."roles"."is_verified", "public"."roles"."company_id" FROM "public"."roles" 
WHERE("public"."roles"."is_verified" = $1 AND "public"."roles"."company_id" = $2 AND 
"public"."roles"."year" = $3 AND 1=0) OFFSET $4 /* traceparent=00-00-00-00 */
```

Note the 1=0

Reply:
nice catch.